lib/Doctrine/ORM/Query/Expr.php

```diff
@@ -607,17 +607,31 @@ private function quoteLiteral($literal)
     }
 
     /**
-     * Creates an instance of BETWEEN() function, with the given argument.
+     * Creates an instance of BETWEEN() function, with the given arguments.
      *
      * @param mixed      $val Valued to be inspected by range values.
      * @param int|string $x   Starting range value to be used in BETWEEN() function.
      * @param int|string $y   End point value to be used in BETWEEN() function.
      *
-     * @return Expr\Func A BETWEEN expression.
+     * @return Expr\Between A BETWEEN expression.
      */
-    public function between($val, $x, $y)
+    public function between($val, $x, $y) : Expr\Between
     {
-        return $val . ' BETWEEN ' . $x . ' AND ' . $y;
+        return new Expr\Between(Expr\Between::BETWEEN, $val, $x, $y);
+    }
+
+    /**
+     * Creates an instance of BETWEEN() function, with the given arguments.
+     *
+     * @param mixed      $val Valued to be inspected by range values.
+     * @param int|string $x   Starting range value to be used in BETWEEN() function.
+     * @param int|string $y   End point value to be used in BETWEEN() function.
+     *
+     * @return Expr\Between A BETWEEN expression.
+     */
+    public function notBetween($val, $x, $y) : Expr\Between
```

Comment:
Would work too.
It would be less complex and easier to read


```php
$qb->expr()->not($qb->expr()->between());

$sql = $qb->getQuery()->getResult();
' .... NOT (1 BETWEEN 1 AND 10) ...
```

```php
$qb->expr()->notBetween();

$sql = $qb->getQuery()->getResult();
' .... 1 NOT BETWEEN 1 AND 10 ...
```

But I can remove it.

Reply:
Yeah, I would skip over the negative version for this patch.